### DISCUSSION TITLE
Logging downloads and all other operations

### DISCUSSION THREAD

**Discussion by brlarini** [filebrowser/filebrowser#2119] [Q&A]
Hi everyone.
While testing Filebrowser, I was checking the way it logs users' file operations.
On this discussion
[https://github.com/filebrowser/filebrowser/issues/1738](https://github.com/filebrowser/filebrowser/issues/1738)
I've seen that logs can (also) be done using the command runner on specific operations (before/after upload, rename, delete, etc).
However it is currently lacking 'before download' and 'after download' events on the config.

Isn't there a way to add commands associated with those events?

Or even an alternative way to log **all** file operations available on command runner, plus downloads (along with username, file, timestamp)?

Thanks!

**Comment by ramiresviana** [comment]
A proxy would allow you to log any request to the server. With some filtering you can get a log of downloads.
